### PR TITLE
guides/http: document Java Jetty client integration

### PR DESCRIPTION
**Pull request by odeke-em**
Document the Java Jetty client integration with a guide,
an end-to-end example and screenshots.

Updates #562

### CODE REVIEW

**Comment by rghetia** [comment]
Looks good. Couple of things.
1. Would it not be better to use same [example](https://github.com/census-instrumentation/opencensus-java/tree/master/examples/src/main/java/io/opencensus/examples/http/jetty) as it is in the repo. If the example in repo is not sufficient then we can enhance or change.
2. An example with client and server both would be better than standalone client.

**Comment by odeke-em** [comment]
Thank you for listing out your concerns @rghetia! Please see my responses below:

>1. Would it not be better to use same example as it is in the repo. If the example in repo is not sufficient then we can enhance or change.

The example in the OpenCensus-Java repo is a full copy and pastable example for both the server and client but here we are documenting step by step for those who just want to use the client integration and the same for those who want to use the server integration. Otherwise, we'd just be dumping for them source code to go inspect. Please see what we've done for Go's net/http https://opencensus.io/guides/http/go/net_http/

> An example with client and server both would be better than standalone client.

Sure for running source code but not for prescriptive steps when users are integrating it to parts of their apps, when someone just wants to focus on the client, them reverse engineering source code yet not everyone runs both client and server.

**Comment by rghetia** [comment]
I didn't mean to dump the source code from the example directory but rather use code snippets from the example directory. Your step by step instructions are great.

**Comment by odeke-em** [comment]
>I didn't mean to dump the source code from the example directory but rather use code snippets from the example directory. Your step by step instructions are great.

Oh I see. You have great examples in the OpenCensus-Java repo, but for this guide, the differences that I made were:
a) not using the log4j logger
b) used two different URLs to demonstrate it capturing status codes: 200 and 404 by fetching from https://opencensus.io/community and https://opencensus.io/non-existent respectively which are then shown in the trace and stats screenshots to demonstrate how the integration enhances and captures observability from requests

File: content/guides/http/java/jetty/client.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+### Imports
+
+To add the Jetty client integration, we'll perform the following import.
+
+{{<highlight Java>}}
+import io.opencensus.contrib.http.jetty.client.OcJettyHttpClient;
+{{</highlight>}}
+
+#### Dependency management
```
Comment (rghetia) [inline content/guides/http/java/jetty/client.md]: what about gradle?

**Comment by rghetia** [comment]
> > I didn't mean to dump the source code from the example directory but rather use code snippets from the example directory. Your step by step instructions are great.
> 
> Oh I see. You have great examples in the OpenCensus-Java repo, but for this guide, the differences that I made were:
> a) not using the log4j logger
> b) used two different URLs to demonstrate it capturing status codes: 200 and 404 by fetching from https://opencensus.io/community and https://opencensus.io/non-existent respectively which are then shown in the trace and stats screenshots to demonstrate how the integration enhances and captures observability from requests

SG.

File: content/guides/http/java/jetty/client.md (hunk 1 above)
Comment (odeke-em) [reply content/guides/http/java/jetty/client.md]: Roger that, I've added:
* Gradle
* Buildr
* Ivy

for the minimum additions of *http-jetty-client and *http-util dependencies.

**Comment by odeke-em** [comment]
Please take a look at the update @rghetia and thanks for the initial review!

**Comment by rghetia** [comment]
LGTM

**Pull request merged**